doc/usage.rst

```diff
@@ -1,6 +1,76 @@
 Usage
 =====
 
+Environment YAML files
```

Comment:
Why did you move this section before the Target section? Especially the support for named resources and drivers is not relevant for most setups and should be explained later.

Reply:
I wanted to unify the seperate explanations and both of the next sections depend on knowledge of the environment yaml configuration syntax. I don't think that seperating the multiple drivers part benefits the reader, we can also explain it to the reader here which also makes it simpler to look up.

Reply:
The library part actually explains how to create Targets, Drivers and Resources manually, so the reader knows how these fit together.

Reply:
@ejoerns we'd like a third opinion on this, please provide yours.

Reply:
The Environment configuration should have an extra chapter to mention all the dirty details and what can be done with it. But I would also agree on @jluebbe's concern that placing it at the very top of the 'usage' chapter leads the user in the wrong direction.
May it will be possible to add short notes wit examples (as done so far) in both the library and the pytest chapter on usage of Environment files and link to a separate chapter for more details. As Environment configs are neither specific to the library nor to the pytest case, I would give it the same level they have as @Emantor already did but maybe place it somewhere below the pytest chapter in doc/usage.rst or simply extend the already existing  'Environment Configuration' chapter in doc/configuration.rst.